### DISCUSSION TITLE
Error deploying contract: invalid private key: invalid length, need 256 bits

### DISCUSSION THREAD

**Discussion by BrwGamer** [gochain/web3#220] [Q&A] [answered]
I'm following this tutorial: https://medium.com/gochain/how-to-create-and-deploy-an-nft-smart-contract-in-10-minutes-e0d7ef8e59fc

When running `sudo web3 contract deploy MYTEST.bin`

This error happens:

`ERROR: Error deploying contract: invalid private key: invalid length, need 256 bits`

I already tried by setting the private key without the `0x` in front of it but the error is the same.

How to solve this?

Thanks

**Comment by treeder** [comment]
Sounds like you don't have `WEB3_PRIVATE_KEY` set and I'd bet it's because
you're running it with `sudo`. You'll have to either set the environment
var for the sudo user or not run it as sudo.

**Reply by BrwGamer** [reply]
Hello,
if I run:
`web3 contract build TEST2.sol`
this error happens:
```
ERROR: Failed to compile "TEST2_flatten.sol": solc: exit status 126
docker: Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/containers/create": dial unix /var/run/docker.sock: connect: permission denied.
See 'docker run --help'.
```

If I do this:
`sudo export WEB3_PRIVATE_KEY=0x...`
This is the error:
`sudo: export: comand not found`

BUT if I set the variabile as explained here (https://stackoverflow.com/a/60939906/2342558), by editing `/etc/enviroment`, then this command works:
`sudo web3 contract deploy TEST3.bin`
but, correctly, it says `ERROR: Error deploying contract: cannot send transaction: insufficient funds for gas * price + value`.

What's the list of the available networks to set in `export WEB3_NETWORK`?
I want to use a free test network.
I tried 'ropsten' but this error occurs:
```
ERROR: Error deploying contract: cannot get gas price: 502 Bad Gateway: <html>
<head><title>502 Bad Gateway</title></head>
<body>
<center><h1>502 Bad Gateway</h1></center>
</body>
</html>
```

Thanks a lot!

**Comment by treeder** [comment]
Use `WEB3_API_URL` and then use whatever rpc server you want. Like one from
infura or something. Or can use GoChain testnet for free.

**Reply by BrwGamer** [reply]
Where can I find a list of all these WEB3_* variables I can set?

**Reply by treeder** [reply]
The README where you'd expect. Or in the cli help.

**Comment by treeder** [comment, marked as answer]
Can you all try to upgrade to v0.2.70 and try again.

**Comment by Jennyfriel** [comment]
How i claim???
